src/herbpy/tsr/box.py

```diff
@@ -0,0 +1,45 @@
+import numpy
+import prpy.tsr
+
+@prpy.tsr.tsrlibrary.TSRFactory('herb', 'box', 'stamp')
+def box_stamp(robot, box, manip=None):
+
+    '''
+    This creates a TSR for stamping a box. 
+    It is assumed that when called, the robot is grasping a stamp
+
+    @param robot The robot to perform the stamp
+    @param box The box to stamp
+    @param manip The manipulator to stamp 
+    '''
+
+    if manip is None:
+        manip_idx = robot.GetActiveManipulatorIndex()
+    else:
+        with manip.GetRobot():
```

Comment:
In relation to this question and the question in which you require to lock, do you mean this?
 
    with robot.GetEnv():
        with robot.CreateRobotStateSaver():
            if manip is None:
                manip_idx = robot.GetActiveManipulatorIndex()
                manip = robot.GetActiveManipulator()
            else:
                manip.SetActive()
                manip_idx = manip.GetRobot().GetActiveManipulatorIndex()

        T0_w = box.GetTransform()